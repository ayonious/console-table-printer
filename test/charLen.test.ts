import { Table } from '../index';

describe('Example: Print a simple Table with Custom char len', () => {
  it('Custom char len is working', () => {
    // Create a table
    const p = new Table({
      charLength: {
        '👋': 2,
        '😅': 2,
        '🚌': 2,
        '👩‍👩‍👧': 2,
      },
    });

    // add rows with color
    p.addRows([
      // adding multiple rows are possible
      {
        Description: '👩‍👩‍👧',
        emoji: '😅',
      },
    ]);

    // print
    p.printTable();
    expect(p.render()).toMatchSnapshot();
  });
});
