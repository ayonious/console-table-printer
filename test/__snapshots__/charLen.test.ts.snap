// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example: Print a simple Table with Custom char len Custom char len is working 1`] = `
"┌─────────────┬───────┐
│[37m [0m[01mDescription[0m │[37m [0m[01memoji[0m │
├─────────────┼───────┤
│[37m [0m[37m         👩‍👩‍👧[0m │[37m [0m[37m   😅[0m │
└─────────────┴───────┘"
`;

exports[`Example: Print a simple Table with cell colors foreign alphabets are working 1`] = `
"┌─────────────┬───────┐
│[37m [0m[01mDescription[0m │[37m [0m[01memoji[0m │
├─────────────┼───────┤
│[37m [0m[37m         👩‍👩‍👧[0m │[37m [0m[37m   😅[0m │
└─────────────┴───────┘"
`;
